sycl/test-e2e/KernelAndProgram/disable-caching.cpp

```diff
@@ -88,6 +88,8 @@ int main() {
 
 // (Program cache releases)
 // CHECK-CACHE: piKernelRelease
-// CHECK-CACHE: piProgramRelease
 // CHECK-CACHE: piKernelRelease
+// CHECK-CACHE: piKernelRelease
+// CHECK-CACHE: piProgramRelease
+// CHECK-CACHE: piProgramRelease
```

Comment:
The cache is destructed in a different order now, which is why these are swapped. Also the previous version of this test only had 2 checks for release of program/kernel even though there were 3 programs/kernels that should have been created, which is why any extra check was added.